### DISCUSSION TITLE
Adding a middleware causes background tasks cancellation

### DISCUSSION THREAD

**Discussion by csuriano23** [fastapi/fastapi#6177] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import asyncio
import time
from fastapi import BackgroundTasks, FastAPI

app = FastAPI()


@app.middleware("http")
async def pass_all_middleware(request, call_next):
    return await call_next(request)


async def test_bg():
    try:
        print("BG TASK STARTED")
        await asyncio.sleep(5)
        print("BG TASK ENDED")
    except BaseException as be:
        print("BG TASK NOT ENDED CORRECTLY", type(be))


@app.get("/hello")
async def root(background_tasks: BackgroundTasks):
    background_tasks.add_task(test_bg)
    return {"message": "hello"}
```

### Description

Executing the sample code causes an unexpected (and apparently unrelated) asyncio `CancelledError` on long running background tasks.

Sample output running on uvicorn and calling GET on `/hello` endpoint:

```text
INFO:     Started server process [95599]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:57617 - "GET /hello HTTP/1.1" 200 OK
BG TASK STARTED
BG TASK NOT ENDED CORRECTLY <class 'asyncio.exceptions.CancelledError'>
```

The cause seems to be the introduction of a middleware; when removing the middleware fragment everything get fine:

```text
INFO:     Started server process [96777]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     127.0.0.1:57674 - "GET /hello HTTP/1.1" 200 OK
BG TASK STARTED
BG TASK ENDED
```

### Operating System

macOS Ventura

### Operating System Details

Darwin ARM64 M1

### FastAPI Version

0.86.0

### Python Version

Python 3.10.8

**Comment by odiseo0** [comment]
I believe this is an issue with `BaseHTTPMiddleware` from Starlette

See [BaseHTTPMiddleware](https://www.starlette.io/middleware/#:~:text=to%20500.-,BaseHTTPMiddleware,-An%20abstract%20class) and this [issue](https://github.com/encode/starlette/issues/1438)

Possible solution:
Use a pure ASGI middleware

**Comment by csuriano23** [comment, marked as answer]
Thank you. It seems that the issue has been fixed in starlette 0.21, so I'm looking forward to have the release with this https://github.com/tiangolo/fastapi/pull/5471

**Comment by csuriano23** [comment]
I think this can be closed after latest FastAPI release. Thank you

**Comment by tiangolo** [comment]
Thanks for the help here @odiseo0 ! 👏  🙇 

Thanks for reporting back and closing the issue @csuriano23  👍
